src/otx/core/data/entity/base.py

```diff
@@ -607,9 +608,19 @@ def collate_fn(
         images = [entity.image for entity in entities]
         like = next(iter(images))
 
+        if stack_images and not all(like.shape == entity.image.shape for entity in entities):  # type: ignore[union-attr]
+            msg = (
+                "You set stack_images as True, but not all images in the batch has same shape. "
+                "In this case, we cannot stack images. Some tasks, e.g., detection, "
+                "can have different image shapes among samples in the batch. However, if it is not your intention, "
+                "consider setting stack_images as False in the config."
+            )
+            warnings.warn(msg, stacklevel=1)
+            stack_images = False
+
         return OTXBatchDataEntity(
             batch_size=batch_size,
-            images=tv_tensors.wrap(cls.stack_batch(images), like=like) if stack_images else images,
+            images=tv_tensors.wrap(torch.stack(images), like=like) if stack_images else images,
```

Comment:
Sorry, I was failed to understand why this is better. When it comes to me, it was possible to configure the variables through `cls.stack_batch(images, pad_size_divisor=..., pad_value=...)`. Please correct me if I was wrong.

Reply:
Yes it could be, but it requires change to data entity of entire tasks. I think it is too risky w.r.t release schedule